### PR TITLE
Fix audio button rounding and global audio not closing

### LINKED ISSUE #1373: Cannot close global audio player after clicking on the waveform
## Description

Global audio player close button becomes not interactive after you've clicked on the waveform.

## Reproduction

1. Open the global audio player.
2. Click on the waveform.
3. Try closing the global audio player by clicking on the x button. It doesn't work.

## Environment

 - Device: mobile and desktop.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #1427: Audio control with rounded corners
## Description
The [audio control component](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=558%3A2289) within the global audio player has rounded corners, creating small gaps in the UI. Reported by @sarayourfriend 

In global audio player, the component should not have rounded corners.

## Reproduction
1. Search any audio result on a small breakpoint where global audio player is visible.
2. Click on play on any audio result.
3. See the rounded corners in the play/pause button in the global audio player.

## Screenshots
CleanShot 2022-05-12 at 15 49 52@2x

## Resolution
- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1427 by @panchovm
Fixes #1373 by @obulat

## Description

This PR adds `connections` prop to `VButton` to enable setting non-rounded corners if the button is a part of a compound control and only needs rounded corners at start/end. So, for a row audio track, we can set `connections=end` and make the button non-rounded where it is connected to the waveform.

Other changes in this PR:
- Clean up some `GlobalAudioTrack` props/attrs that were not used before, or fell out of sync with the base `AudioTrack` component (e.g. `ariaLabel` wasn't computed).
- Set the `GlobalAudioTrack` `Close` button `z-index` to 20 to prevent the waveform from overlapping. This fixes #1373. It's a one-character change, so I added it in here instead of opening a separate PR.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VAudioTrack/VPlayPause.vue [hunk 1]
```diff
@@ -69,12 +70,27 @@ export default defineComponent({
...
      */
     const icon = computed(() => statusIconMap[props.status])
+
+    const buttonProps = computed(() => {
+      const connections = (
+        props.layout === 'row'
+          ? 'end'
+          : props.layout === 'global'
+          ? 'all'
+          : 'none'
```
Comment (dhruvkb) [inline src/components/VAudioTrack/VPlayPause.vue:80]: The chained ternaries are a tad bit hard to read.

**Review by dhruvkb — APPROVED** [review]
This is a good fix, very clever use of connections.

File: src/components/VAudioTrack/VPlayPause.vue (hunk 1 above)
Comment (obulat) [reply src/components/VAudioTrack/VPlayPause.vue:80]: I don't like it either, extracted it to a mapping object.

**Comment by obulat** [comment]
> This is a good fix, very clever use of connections.

Thank you, @dhruvkb, I actually used your code from the Input component :)

**Pull request merged**